### PR TITLE
feat: get_node early return

### PR DESCRIPTION
**Pull request by lightsing**
merge #31 before this

### CODE REVIEW

**Comment by lispc** [comment]
I am not too familiar with the codes here.   So to make reviewing easier, we can split this pr into 2 parts: first one: only add the MAX_LEVELS to constant generic but not change the behavior at all, part 2 fix the issue.   I guess part 2 Will be not too many lines.

**Comment by lightsing** [comment]
const generic is split into #31

File: rs_zktrie/src/raw.rs [hunk 1]
```diff
@@ -352,68 +352,115 @@ impl<H: Hashable, DB: ZktrieDatabase, const MAX_LEVELS: usize> ZkTrieImpl<H, DB,
...
-                _ => Err(ImplError::ErrInvalidNodeFound),
-            }?
+                _ => return Err(ImplError::ErrInvalidNodeFound),
+            }
         }
-        Ok((node.unwrap(), siblings))
+        Err(ImplError::ErrKeyNotFound)
     }
 
+    // fn try_get_with_path(&self, node_key: &H) -> Result<(Node<H>, Vec<H>), ImplError> {
```
Comment (noel2004) [inline rs_zktrie/src/raw.rs:405]: The optimization is good. Just consider maybe we should keep this old function with another name.

It return both the found node and the path, which is important for mpt proof.

Currently the rs-zktrie do not be used by `l2geth` but i am not sure what would be our future plan like.

**Review by noel2004 — APPROVED** [review]

File: rs_zktrie/src/raw.rs (hunk 1 above)
Comment (lightsing) [reply rs_zktrie/src/raw.rs:405]: so I comment out this, since it's a private method and no usage.

Comment (noel2004) [reply rs_zktrie/src/raw.rs:405]: It is ok. What about just leave some comment so we would not prune it wrongly later?

**Pull request merged**